### ISSUE TITLE
Treat all text between /* eslint-disable */ and /* eslint-enable */ as opaque text

### ISSUE THREAD

**Issue by bedney** [eslint/eslint#2428]
I recently ran into a use case where I use Handlebars to template values into a JS file. This gives a syntax error, as expected:

```
TP.boot.$bootversion = {{bootversion}};
```

Even surrounding this text with `/* eslint-disable */` and `/* eslint-enable */ ` doesn't cause the warning to go away, which is unexpected. I'd like to leave this as a `.js` file, to take advantage of syntax-coloring editors & other tools.

In reading issue #900, I realized that eslint invokes Espree directly. Is there a way to add a level of indirection such that, as soon as  `/* eslint-disable */` is encountered, the eslint engine treats all bytes as opaque until `/* eslint-enable */ ` is encountered again?

This would solve a variety of issues, including having Handlebars, JSX (which I know now eslint has other solutions for) and any other thing that users might want to introduce into their `.js` files.

**Comment by ilyavolodin** [comment]
Unfortunately, that would be extremely hard to do. ESLint has pluggable interface for parsers. You can use espree (by default), esprima, babble-eslint, etc. Each and every one of them has to support those comments, since eslint only deals with comments after it gets AST back from the parser. I suggest you use .eslintignore file to just disable linting for the files in question.
You can also write a custom processor (http://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins) to remove offending text before sending it to eslint. You can even make a generic processor, that would find enable/disable comment and remove the code in between of them.

**Comment by bedney** [comment]
A generic processor looks like the way to go here. Is there any reason why that wouldn't be considered a 'built-in' processor for eslint in general? i.e. a processor that would automatically strip the content between `/* eslint-disable */` and `/* eslint-enable */` and would just automatically be run as part of eslint. IMHO, it would meet the semantic definition of those two flags (i.e. ignore all bytes between these two flags). I'm sure there are other opinions regarding the semantic meaning of these flags though ;-).

**Comment by nzakas** [comment]
Because `/*eslint-disable*/` can also be used to disable just one rule, and the purpose of the comment is to disable *rules* not *parsing*.

If you create a plugin for this, I'd suggest using a different comment to mark the area not to parse.

This is outside the scope of ESLint's core responsibilities, so it would be up to you to create a plugin.
